keps/754-time-dependent-job-priority/README.md

```diff
@@ -0,0 +1,197 @@
+  - [Other points](#other-points)
+  - [Test Plan](#test-plan)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+This proposal allows job priorities to be dynamically adjusted based on time.  
+This functionality aims to prioritize the execution of jobs that have been waiting 
+for an extended period, even if they have a lower initial priority.  
+By introducing time-dependent priority adjustments, the system can ensure that jobs 
+that have been deferred receive the attention they require.  
+This KEP outlines the necessary changes and considerations to incorporate this 
+time-based priority mechanism.
+
+## Motivation
+
+Lower-priority jobs sometimes get postponed for long time than expected.  
+To address situations where jobs remain unexecuted even after a certain period of time,
+we implementate this KEP.
+
+### Goals
+
+Provide a mechanism where the priority of a job increase after a certain period of
+time has elapsed.
+
+### Non-Goals
+
+The timed out job is going to the head of the ClusterQueue instead of changing 
+the priority.
+
+## Proposal
+
+Replace the priority of Jobs with a calculation that takes into account the time since the Job was created.  
+This replacement will modify the calculation methods for Preemption and Queueing strategy.  
+The priority of Job that don't have `WorkloadPriorityClass` will remain unchanged.
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories
+
+#### Story 1
+
+The job doesn't have a high priority but is intended for production, so it needs to start execution within three hours.
+
+
+### Risks and Mitigations
+
+If the job is just popped out for scheduling, the running job would be preempted easily by other Jobs.  
+That's why this mechanism should be based on priority of job.
+
+## Design Details
+
+### Kueue WorkloadPriorityClass API
+
+We introduce the Kueue `WorkloadPriorityClass` API.
+
+```golang
+// WorkloadPriorityClass defines basePriority and time dependent calculation logic
+type WorkloadPriorityClass struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   WorkloadPrioritySpec   `json:"spec,omitempty"`  
+}
+
+type WorkloadPrioritySpec struct {
+  // initial priority of workload
+	Priority int32 `json:"priority,omitempty"`
+
+  // +optional
+  DynamicPriorityPolicy DynamicPriorityPolicy `json:"dynamicPriorityPolicy,omitempty"`
+
+}
+
+type DynamicPriorityPolicy struct {
```

Comment:
This can quickly become either insufficient or very cluttered. For example, I may want to have this logic outside of business hours and on weekends. Plus maybe deprioritise workloads that have been waiting for too long as their results may not actually be that relevant anymore.  And so on.